iothub/device/src/IotHubConnectionString.cs

```diff
@@ -75,6 +84,10 @@ public IotHubConnectionString(IotHubConnectionStringBuilder builder)
 
                 Debug.Assert(TokenRefresher != null);
             }
+            else if (!string.IsNullOrWhiteSpace(builder.SharedAccessSignature))
+            {
+                SharedAccessSignature = builder.SharedAccessSignature;
```

Comment:
If builder.SharedAccessSignature is null, SharedAccessSignature will remain the default null. Why not just do, SharedAccessSignature = builder.SharedAccessSignature like it was earlier?

Reply:
The reason I moved it here is because the SharedAccessSignature should be set only if the there isn't an AuthenticationWithTokenRefresh authentication method already supplied to the client; i.e. it is placed in the `else` clase.

Reasoning:
"IotHubConnectionString.SharedAccessSignature should be set only if it is non-null and the authentication method of the device client is not of type AuthenticationWithTokenRefresh. Setting the sas value for an AuthenticationWithTokenRefresh authentication type will result in tokens not being renewed. This flow can be hit if the same authentication method is always used to initialize the client; as in, on disposal and reinitialization. (the value of the sas token computed is stored within the authentication method, and on reinitialization the client is incorrectly identified as a fixed-sas-token-initialized client, instead of being identified as a sas-token-refresh-enabled-client.)"

I'll add this to the code comments, just so we don't lose the context in the future.